homeassistant/helpers/config_validation.py

```diff
@@ -376,9 +376,12 @@ def validator(value):
         """Validate ordered dict."""
         config = OrderedDict()
 
-        for key, val in value.items():
-            v_res = item_validator({key: val})
-            config.update(v_res)
+        try:
+            for key, val in value.items():
+                v_res = item_validator({key: val})
+                config.update(v_res)
+        except AttributeError:
```

Comment:
This cannot happen. We create a new dictionary on line 381 which is passed into 382. That's always a dictionary.